### PR TITLE
Add note about not enforcing stream limit on first mount.

### PR DESCRIPTION
**Pull request by crova**
# Comment
From our slack discussion, here is a PR to warn folks about stream limits not being enforced on the first render.
Let me know if you see any issues.

Cheers.

### CODE REVIEW

File: lib/phoenix_live_view.ex [hunk 1]
```diff
@@ -1602,6 +1602,10 @@ defmodule Phoenix.LiveView do
   positive limit will prune items from the end of the container, while a negative
   limit will prune items from the beginning of the container.
 
+  Note that on the first `mount/3` render (when no websocket connection was established yet),
+  the limit set is not enforced. In this situation, one should pass the desired amount of
+  items to the stream to avoid confusions.
```
Comment (josevalim) [inline lib/phoenix_live_view.ex:1607]:
```suggestion
  Note that the limit is not enforced on the first `mount/3` render (when no websocket
  connection was established yet), as it means more data than necessary has been
  loaded. In such cases, you should only load and pass the desired amount of items
  to the stream.
```

**Comment by josevalim** [comment]
I dropped a suggestion but I wonder if the code should also warn in those cases. Then we may choose to also limit it or not (the advantage of not limiting is that it may be easier to spot, at the cost of wrong behaviour, so I am not sure).

**Comment by crova** [comment]
> I dropped a suggestion 

Your wording is much cleaner than mine, cheers.

> but I wonder if the code should also warn in those cases. Then we may choose to also limit it or not (the advantage of not limiting is that it may be easier to spot, at the cost of wrong behaviour, so I am not sure).

A warn sounds nice to me.
Chris was not sure if LV should enforce it and let users pass more data than needed. I tend to agree with the idea that we should pass the desired amount of data, but a warning don't seem too far fetched here.

**Comment by greven** [comment]
Just want to add that I was exactly hunting down why wouldn't limit work on first render, didn't see anything in the docs so this is very useful addition.

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Pull request merged**